tools/confedit/src/main.rs

```diff
@@ -0,0 +1,128 @@
+use std::{borrow::Cow, io::{Read, Write}};
+use clap::clap_app;
+use loadstone_config::{Configuration, features::{Greetings, Serial}, security::{SecurityConfiguration, SecurityMode}};
+
+struct Arguments {
+    greeting: Option<String>,
+    golden_bank: Option<Option<usize>>,
+    recovery: Option<bool>,
+}
+
+fn read_input_string() -> Result<String, String> {
+    let mut input = String::default();
+    let result = std::io::stdin().read_to_string(&mut input);
+    match result {
+        Ok(_) => Ok(input),
+        Err(e) => Err(format!("failed to read from standard input stream: {}.", e)),
+    }
+}
+
+fn get_input_configuration(string: String) -> Result<Configuration, String> {
+    if string == "" {
+        Ok(Configuration {
+            security_configuration: SecurityConfiguration {
+                security_mode: SecurityMode::Crc,
+                ..SecurityConfiguration::default()
+            },
+            ..Configuration::default()
+        })
+    } else {
+        ron::from_str(&string)
+            .map_err(|e| format!("failed to load configuration from input: {}.", e))
+    }
+}
+
+fn modify_configuration(mut configuration: Configuration, arguments: Arguments) -> Result<Configuration, String> {
+    if let Some(greeting) = arguments.greeting {
+        let old_demo = match configuration.feature_configuration.greetings {
+            Greetings::Default => Cow::from(""),
+            Greetings::Custom { demo, .. } => demo,
+        };
+
+        configuration.feature_configuration.greetings = Greetings::Custom {
+            loadstone: Cow::from(greeting),
+            demo: old_demo,
+        };
+    }
+
+    if let Some(bank) = arguments.golden_bank {
+        configuration.memory_configuration.golden_index = bank;
+    }
+
+    if let Some(recovery) = arguments.recovery {
+        let serial = &mut configuration.feature_configuration.serial;
+        if let Serial::Enabled { recovery_enabled, .. } = serial {
+            *recovery_enabled = recovery;
+        } else {
+            return Err(String::from("cannot enable serial recovery since serial is not enabled"));
+        }
+    }
+
+    Ok(configuration)
+}
+
+fn get_output_string(configuration: Configuration) -> Result<String, String> {
```

Comment:
It's not a big deal for a quick script like this, but using the `anyhow` crate would make the error management a lot easier. Just return an `anyhow::Result` and you get to use `?` everywhere instead of `map_err`